Automattic/Gravatar-SDK-iOS
# Handle payload too large error on upload

Closes #436 

### Description

For the reasons explained in  #436 we need to handle 413(payloadTooLarge) error on the client side to show the proper message.

### Testing Steps

You need to manipulate the code and set `maxOutputImageSizeInPixels` to a value bigger than 20000.
Again manipulate the code to upload png instead of jpeg.

You can try with `32MBImage.png` from https://drive.google.com/drive/u/0/folders/16bRuYugP99X3grDlh938ffJaomlfFXaT
Try to upload it and you should see the error "The provided image exceeds the maximum size: 10MB"

Regression test:

Try again with `20008x20008.png` from the same folder.
Observe that it gives you the dimension error this time.
